api/nodes/serializers.py

```diff
@@ -464,6 +467,19 @@ class NodeSerializer(TaxonomizableSerializerMixin, JSONAPISerializer):
     ))
 
     def get_current_user_permissions(self, obj):
+        request = self.context['request']
+        if StrictVersion(request.version) < StrictVersion('2.11'):
+            return self.get_current_user_permissions_legacy(obj)
+        user = self.context['request'].user
+        if isinstance(user, AnonymousUser):
+            perms = []
+        else:
+            perms = obj.get_permissions(user)
+            if user in obj.parent_admin_contributors:
+                perms += ['read']
+        return perms
+
+    def get_current_user_permissions_legacy(self, obj):
         if hasattr(obj, 'contrib_admin'):
```

Comment:
Do you know what this "contrib_admin" and "contrib_write" business is? Looks like an annotation optimization so we don't have to perform `obj.get_permission(user)`. Should this logic be included in the new function?

Reply:
Extremely good point! Re-worked a few things to include this again.

Note: in my re-working, I realize that this does indeed double check for `contrib_admin` on `obj`. I tried many times to remove this double check, but in the end it was nicer to keep it instead of doing too much refactoring on how the legacy endpoint logic was working.